pkg/model/manifest.go

```diff
@@ -422,11 +422,23 @@ func (m Manifest) ManifestName() ManifestName {
 	return m.Name
 }
 
-func LocalRefSelectorsForManifests(manifests []Manifest) []container.RefSelector {
+func LocalRefSelectorsForManifests(manifests []Manifest, clusters map[string]*v1alpha1.Cluster) []container.RefSelector {
 	var res []container.RefSelector
 	for _, m := range manifests {
+		cluster := clusters[m.ClusterName()]
+		if m.ClusterName() != "" && cluster == nil {
```

Comment:
this 'if' branch is kind of confusing, is it necessary? seems like everything should be ok if the cluster is nil

Reply:
I got rid of it - it was trying to skip any targets where we were missing the `Cluster` object since we otherwise might produce the wrong ref (i.e. w/o local registry). That said, it's harmless if it attempts and arguably not any less confusing than silently skipping it, so I don't feel bad about nixing it 🙃